scripts/triggers_vm_generator.py

```diff
@@ -0,0 +1,199 @@
+      ])
+
+    included_files = self._remove_duplicates(included_files)
+    return included_files
+
+  def _remove_duplicates(self, included_files):
+    """Removes duplicates from a List."""
+    final_list = []
+    for num in included_files:
+      if num not in final_list:
+        final_list.append(num)
+    return final_list
+
+  def generate_config(self, included_files):
+    """Generates GCB trigger config."""
+    included_files.sort()
+    trigger = {
+        'description': 'Trigger for VM %s' % self._solution,
+        'filename': CLOUDBUILD_CONFIG_FILE,
+        'github': {
+            'name': 'click-to-deploy',
+            'owner': 'GoogleCloudPlatform',
+            'pullRequest': {
+                'branch': '.*',
+                'commentControl': 'COMMENTS_ENABLED'
+            }
+        },
+        'includedFiles': included_files,
+        'substitutions': {
+            '_SOLUTION_NAME': self._solution
+        }
+    }
+    return trigger
+
+
+class CreateThreadPoolAndWait(object):
+  """Creates thread pool and wait for all jobs to finish.
+
+  For example:
+
+  with CreateThreadPoolAndWait() as pool:
+    result1=pool.apply_async(func1)
+    result2=pool.apply_async(func2)
+  """
+
+  def __init__(self):
+    self._pool = multiprocessing.pool.ThreadPool()
+
+  def __enter__(self):
+    return self._pool
+
+  def __exit__(self, exc_type, exc_val, exc_tb):
+    self._pool.close()
+    self._pool.join()
+
+
+def invoke_shell(args):
+  """Invokes a shell command."""
+  logging.debug('Executing command: %s', args)
+  child = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+  output, _ = child.communicate()
+  exit_code = child.returncode
+  return output, exit_code
+
+
+def get_cookbook_deps(cookbook, knife_binary):
+  """Returns cookbooks dependencies."""
+  if cookbook in _COOKBOOKS:
+    # do not check cookbook twice
+    return _COOKBOOKS[cookbook]
+
+  command = [
+      knife_binary, 'deps', '--config-option',
+      'cookbook_path=%s' % COOKBOOKS_DIR,
+      os.path.join('/cookbooks', cookbook)
+  ]
+  deps, exit_code = invoke_shell(command)
+  assert exit_code == 0, exit_code
+  deps = [dep.replace('/cookbooks/', '') for dep in deps.splitlines()]
+
+  _COOKBOOKS[cookbook] = deps
+  return deps
+
+
+def get_solutions_list():
+  """Returns list of solutions."""
+  listdir = [
+      f for f in os.listdir(PACKER_DIR)
+      if os.path.isdir(os.path.join(PACKER_DIR, f))
+  ]
+  listdir.sort()
+  return listdir
+
+
+def generate_config(solution, knife_binary):
+  trigger = VmTriggerConfig(solution=solution, knife_binary=knife_binary)
+  included_files = trigger.included_files
+  return trigger.generate_config(included_files)
+
+
+def main():
+  parser = argparse.ArgumentParser()
+  parser.add_argument(
+      '--knife_binary', type=str, default='knife', help='knife-solo binary')
+  args = parser.parse_args()
+
+  listdir = get_solutions_list()
+
+  with CreateThreadPoolAndWait() as pool:
```

Comment:
seems overkill--the `generate_config` is not bounded by I/O and doesn't need to be parallelized?

Reply:
`knife` command takes about 30 seconds to finish. Thanks to this, we can save some time.